packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2528,17 +2528,13 @@ export class ContainerRuntime
 			return;
 		}
 
-		// We need to temporary clear the dirty flags and disable
-		// dirty state change events to detect whether replaying ops
-		// has any effect.
-
-		// Save the old state, reset to false, disable event emit
-		const oldState = this.dirtyContainer;
-		this.dirtyContainer = false;
-
+		// Replaying is an internal operation and we don't want to generate noise while doing it.
```

Comment:
do we really need this? in the updateDocumentDirtyState we only emit if the state changes, won't the state just go from dirty to dirty, so nothing will be emitted?

Reply:
ideally we could get rid of emitDirtyDocumentEvent too

Reply:
Hmmm I would like to get rid of this too, and you make a good point.

If it's not dirty now, this will be a big no-op.  And the calls to `updateDocumentDirtyState` which can transition to "saved" can't be reached while replaying pending ops.

So, yeah look safe.

Reply:
I think I am going to do that in a follow-up PR, it feels more bold of a "refactor" than I'm going for in this one.

Reply:
seems fine, but current change does a bunch of refactoring already.